### PR TITLE
[test] Client-side controller tests (admin, navigation, pages)

### PR DESCRIPTION
**Pull request by marcoviappiani**
#115

this contains unit tests for the following controllers:
- [x] Admin
- [x] Navigation
- [x] Pages

### CODE REVIEW

**Comment by RRDuke** [comment]
LGTM

File: modules/core/tests/client/navigationController.test.client.js [hunk 1]
```diff
@@ -1,17 +1,108 @@
...
+    });
+
+    it('should move to the signup page after launching signup', function () {
+      spyOn($state, 'go');
+      vm.signup();
+      expect($state.go).toHaveBeenCalledWith('signup');
+    });
+  });
+
+  describe('user', function (){
```
Comment (codydaig) [inline modules/core/tests/client/navigationController.test.client.js]: Space

File: modules/core/tests/client/navigationController.test.client.js [hunk 2]
```diff
@@ -1,17 +1,108 @@
...
+    });
+  });
+
+  describe('logout', function (){
+    it('logout should be a function', function () {
+      expect(typeof vm.logout).toEqual('function');
+    });
+  });
+
+  describe('signup', function (){
```
Comment (codydaig) [inline modules/core/tests/client/navigationController.test.client.js]: Space

File: modules/core/tests/client/navigationController.test.client.js [hunk 3]
```diff
@@ -1,17 +1,108 @@
...
+    });
+
+    it('should move to the login page after launching login', function () {
+      spyOn($state, 'go');
+      vm.login();
+      expect($state.go).toHaveBeenCalledWith('login');
+    });
+  });
+
+  describe('logout', function (){
```
Comment (codydaig) [inline modules/core/tests/client/navigationController.test.client.js]: Space

File: modules/core/tests/client/navigationController.test.client.js [hunk 4]
```diff
@@ -1,17 +1,108 @@
...
+    });
+  });
+
+  describe('tabs', function (){
+    it('tabs should be an array', function () {
+      expect(Array.isArray(vm.tabs)).toBe(true);
+    });
+  });
+
+  describe('login', function (){
```
Comment (codydaig) [inline modules/core/tests/client/navigationController.test.client.js]: Space

File: modules/core/tests/client/navigationController.test.client.js [hunk 5]
```diff
@@ -1,17 +1,108 @@
...
+    it('vm should have a login property', function () {
+      expect(vm.login).toBeDefined();
+    });
+
+    it('vm should have a signup property', function () {
+      expect(vm.signup).toBeDefined();
+    });
+  });
+
+  describe('tabs', function (){
```
Comment (codydaig) [inline modules/core/tests/client/navigationController.test.client.js]: Space

**Pull request merged**